pkg/operator/mon/health.go

```
@@ -0,0 +1,116 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mon
+
+import (
+	"fmt"
+
+	"github.com/rook/rook/pkg/cephmgr/client"
+	"github.com/rook/rook/pkg/cephmgr/mon"
+	"github.com/rook/rook/pkg/clusterd"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (c *Cluster) CheckHealth() error {
+	logger.Debugf("Checking health for mons. %+v", c.clusterInfo)
+
+	// connect to the mons
+	ctx := &clusterd.Context{ConfigDir: c.configDir}
+	conn, err := mon.ConnectToClusterAsAdmin(ctx, c.context.Factory, c.clusterInfo)
+	if err != nil {
+		return fmt.Errorf("cannot connect to cluster. %+v", err)
+	}
+	defer conn.Shutdown()
+
+	// get the status and check for quorum
+	status, err := client.GetMonStatus(conn)
+	if err != nil {
+		return fmt.Errorf("failed to get mon status. %+v", err)
+	}
+	logger.Debugf("Mon status: %+v", status)
+
+	// failover the unhealthy mons
+	for _, mon := range status.MonMap.Mons {
+		inQuorum := monInQuorum(mon, status.Quorum)
+		if inQuorum {
+			logger.Debugf("mon %s found in quorum", mon.Name)
+		} else {
+			logger.Warningf("mon %s NOT found in quorum. %+v", mon.Name, status)
+
+			if len(status.MonMap.Mons) > c.Size {
+				// no need to create a new mon since we have an extra
+				err = c.removeMon(conn, mon.Name)
+			} else {
+				// bring up a new mon to replace the unhealthy mon
+				err = c.failoverMon(conn, mon.Name)
+			}
+			if err != nil {
+				logger.Errorf("failed to failover mon %s. %+v", mon.Name, err)
```

Dont you want to return this error?